### ISSUE TITLE
The dependence package "on-change" need updating.

### ISSUE THREAD

**Issue by TanShun** [smithki/storage-proxy#19]
The files in the package installed by npm are only dist files,  it seems that the dependence package "on-change" isn' t the latest one. It can't save the values to the storage by using functions, for example: 
```js
const myLocalStorage = StorageProxy.createLocalStorage('my-namespace');
myLocalStorage.foo = [1, 2, 3];
myLocalStorage.foo.push(4);//  The value in the localStorage is still [1, 2, 3], not [1, 2, 3, 4]
```

**Comment by smithki** [comment]
Hmmm... I cannot seem to reproduce the issue you are experiencing with mutable array methods  not updating the values correctly. Just to be safe, I am adding coverage for a few commonly used array methods and updating `on-change` to `^1.6.2`...

Do you have a code sample that reproduces the issue so I can dig deeper?

**Comment by smithki** [comment]
> The files in the package installed by npm are only dist files

This is intentional. I am using `microbundle` to package the code for browser environments. The source code will always be available for reference here on GitHub!

**Comment by smithki** [comment]
Updated `on-change` has landed in `v0.11.3`. Please update your version and please open a new issue if any bugs persist. Feel free to continue this thread if you need support!
